Doc/tools/version_next.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+"""
+Replace `.. versionchanged:: next` lines in docs files by the given version.
+
+Run this at release time to replace `next` with the just-released version
+in the sources.
+
+No backups are made; add/commit to Git before running the script.
+
+Applies to all the VersionChange directives. For deprecated-removed, only
+handle the first argument (deprecation version, not the removal version).
+
+"""
+
+import re
+import sys
+import argparse
+from pathlib import Path
+
+DIRECTIVE_RE = re.compile(
+    r'''
+        (?P<before>
+            \s*\.\.\s+
+            (version(added|changed|removed)|deprecated(-removed)?)
+            \s*::\s*
+        )
+        next
+        (?P<after>
+            .*
+        )
+    ''',
+    re.VERBOSE | re.DOTALL,
+)
+
+docdir = (Path(__file__)
+          .parent  # cpython/Doc/tools
+          .parent  # cpython/Doc
+          .resolve()
+          )
+
+parser = argparse.ArgumentParser(
+    description=__doc__,
+    formatter_class=argparse.RawDescriptionHelpFormatter)
+parser.add_argument('version',
+                    help='String to replace "next" with.')
+parser.add_argument('directory', type=Path, nargs='?',
+                    help=f'Directory to process. Default: {docdir}',
+                    default=docdir)
+parser.add_argument('--verbose', '-v', action='count', default=0)
+
+def main(argv):
+    args = parser.parse_args(argv)
+    version = args.version
+    if args.verbose:
+        print(
+            f'Updating "next" versions in {args.directory} to {version!r}',
+            file=sys.stderr)
+    for path in Path(args.directory).glob('**/*.rst'):
```

Comment:
Do you know if/how the tool should be used for translated docs?
Should this line include `*.po` files as well, or run on the converted `*.rst` files?

I don't think the current regex works with arbitrary `*.po` files, since the directive is surrounded by quotes and technically it might be split into two strings.

Reply:
`next` is an identifier that only appears in the source. It shouldn't be translated, and `version_next.py` shouldn't deal with translations.

The user-facing version string is language-agnostic (e.g. `3.13`, `3.14.0a0`) with a translatable suffix: the PR adds `sphinx_gettext('{} (unreleased)').format(release)` to `pyspecific.py`.